### PR TITLE
Feature/footnote popover

### PR DESCRIPTION
**Pull request by yamahito**
Updated footnote popover looks and behavior, for both devices with and without touch.
On devices with touch, the popover shows as a modal from the screen on tapping on the link.
For other devices it shows as a normal popover on hovering over the link.
Footnote popover is now interactive, so the user can select text in them or click links.

### CODE REVIEW

**Comment by yamahito** [comment]
@ccheraa can you please add any documentation, add @joewiz and an internal review, then mark as ready for review.  Thanks!

File: tests/specs/footnote/uat_footnote_popover.spec.js [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+      assert.equal(Page.getElement('.popover .footnote-body a.fn-back').isDisplayed(), false);
+    });
+    it('footnote popover should not disappear with the cursor on it', function () {
+      assert.equal(Page.getElement('.popover').isDisplayed(), false);
+      Page.getElement('#content-inner a.note[href^="#fn\\:"][rel=footnote]').scrollIntoView();
+      Page.getElement('#content-inner a.note[href^="#fn\\:"][rel=footnote]').moveTo();
+      Page.waitForVisible('.popover');
+      assert.equal(Page.getElement('.popover').isDisplayed(), true);
+      Page.getElement('.popover').scrollIntoView();
+      Page.getElement('.popover').moveTo();
```
Comment (plutonik-a) [inline tests/specs/footnote/uat_footnote_popover.spec.js:54]: `moveTo()` without arguments is failing currently.
This is the error:
```
from invalid argument: 'y' must be an int
  (Session info: chrome=104.0.5112.79)
```
That means running Chrom / Chromedriver v104 with current test framework at version `@wdio/cli@5.18.6` is probably not supported.

**Review by plutonik-a — CHANGES_REQUESTED** [review]

**Comment by plutonik-a** [comment]
Footnote number should be shown in the popover.

**Review by plutonik-a — CHANGES_REQUESTED** [review]
See https://github.com/evolvedbinary/hsg-shell/pull/8#issuecomment-1209448912

**Review by plutonik-a — CHANGES_REQUESTED** [review]
I’ve rerun the tests - 11 are still failing. Could you try to fix these by specifying scrollIntoView with scrollIntoView({behavior: "smooth", block: "center", inline: "center"}), see https://github.com/evolvedbinary/hsg-shell/blob/bc1a52c2f46fea05f075692934b1f12cdfd57dc0/tests/specs/UI-components/uat_tooltip.spec.js#L58-L66.

**Review by plutonik-a — APPROVED** [review]

**Pull request merged**